screens/Challenges/ToDoScreen.js

```diff
@@ -12,25 +12,41 @@ import Fonts from "../../constants/Fonts";
 import { getAllAssignedChallenges } from "../../utils/db/challenges";
 import { getUserId } from "../../utils/auth/auth";
 
-export default function ToDoScreen() {
+export default function ToDoScreen(props) {
   const [challenges, setChallenges] = useState([]);
+  const [currentUser, setCurrentUser] = useState("");
 
   function renderTodos() {
     let todos = []
     let alternator = true;
     for (challenge of challenges) {
-      // console.log(challenge)
+      console.log(challenge.assignedChallengeId)
+      if(challenge.submissions && challenge.submissions[currentUser]) continue; //If the user has a submission for this challenge, filter it out from Todo
+      
+      let screentoNav;
+      if(challenge.challengeDetails.mediaType === "STRING"){
+        screentoNav = "UploadText";
+      }
+      else if(challenge.challengeDetails.mediaType === "IMAGE"){
+        screentoNav = "UploadImage";
+      }
+      else{
+        //Do some error handling
+      }
       todos.push(
-        <ChallengeTodo key={challenge.teamId} challenge={challenge} mainColor={(alternator ? Colors.KIT_LIGHT_ORANGE : Colors.KIT_GREEN)} onPress={() => {alert("Hello")}}/>
+        <ChallengeTodo key={challenge.assignedChallengeId} challenge={challenge} mainColor={(alternator ? Colors.KIT_LIGHT_ORANGE : Colors.KIT_GREEN)} onPress={() => {
+          props.navigation.navigate(screentoNav, {challenge})
+        }}/>
       )
       alternator = !alternator;
     }
     return todos
   }
 
   useEffect(() => {
-    currentUser = getUserId();
-    getAllAssignedChallenges(currentUser).then(challenges => {
+    currUser = getUserId();
+    setCurrentUser(currUser);
+    getAllAssignedChallenges(currUser).then(challenges => {
```

Comment:
Just a note that we should eventually change this so it knows to rerender whenever a challenge has been assigned.